### DISCUSSION TITLE
无法安装

### DISCUSSION THREAD

**Discussion by WcZaHr** [ywmoyue/biliuwp-lite#807] [General]
### Bug 描述

Windows11系统，安装证书后打开安装包，安装不成功，显示应用安装程序无法安装包相关项

### 操作步骤

```plain text
安装证书
打开安装包
```

### 期望结果

正常安装更新

### 截图

![image](https://github.com/user-attachments/assets/54b3d96a-7788-4cb6-86ba-696484bf8268)

### 应用来源

Github/其他Github镜像站

### 安装方式

appx直接安装

### 系统版本

其它 (需注明)

### 其它系统版本

Windows11 23H2

### 设备类型

笔记本

### 设备架构

x64

### 应用架构

x64

### 代码贡献

我更擅长发现问题

**Comment by ywmoyue** [comment]
第一次安装请使用脚本方式安装

**Reply by WcZaHr** [reply]
这不是第一次安装，是更新版本

**Reply by ywmoyue** [reply]
@WcZaHr 4.6.x更新4.7.x需要卸载后用脚本方式安装

如果希望保留数据可以先更新4.7.0.102的native版本使用设置页中开发者选项中的导出设置功能

**Reply by WcZaHr** [reply]
无法执行脚本，判断程序返回Restricted，脚本没法用啊

**Reply by ywmoyue** [reply]
@WcZaHr 这是权限问题，安装教程中有说明

**Reply by WcZaHr** [reply]
没找到啊，麻烦指个路谢谢

**Reply by ywmoyue** [reply]
![image](https://github.com/user-attachments/assets/d025b3b4-f58e-4ae9-972c-b3f64e55e0da)

https://github.com/ywmoyue/biliuwp-lite/blob/dev/document/install-readme.md#%E6%A3%80%E6%9F%A5%E8%84%9A%E6%9C%AC%E6%89%A7%E8%A1%8C%E6%9D%83%E9%99%90

**Comment by WcZaHr** [comment]
![image](https://github.com/user-attachments/assets/000937be-17f2-4944-8310-fac7d12db9fa)
执行了这个命令Set-ExecutionPolicy -ExecutionPolicy RemoteSigned，就出现了上面这张图的情况

**Reply by ywmoyue** [reply]
看起来是没有询问是否开启让你输入Y，继续安装教程中的下一步呢

**Comment by WcZaHr** [comment]
![image](https://github.com/user-attachments/assets/bf1312a0-d435-46de-9ca2-4a5f0330adcd)
就是这个样子了

**Reply by ywmoyue** [reply]
下一步是指运行安装脚本https://github.com/ywmoyue/biliuwp-lite/blob/dev/document/install-readme.md#%E8%BF%90%E8%A1%8C%E5%AE%89%E8%A3%85%E8%84%9A%E6%9C%AC

**Reply by WcZaHr** [reply]
可以安装了，但是以后每次都得这样开两个PowerShell来进行安装嘛？
如果直接打开这个安装包的话会显示需要安装什么框架，能否告诉一下这个是什么框架，然后自己安装这个框架，就可以直接安装这个应用了？

Uploading 屏幕录制 2024-09-15 154550.mp4…

**Reply by WcZaHr** [reply]
![image](https://github.com/user-attachments/assets/02ebe533-1b70-423c-ac6a-bdaa7dec0d73)
大概是这样的

**Reply by ywmoyue** [reply]
直接打开这个安装包会从应用商店安装依赖框架，但这个框架安装包已经在解压目录中的Dependencies目录下，运行安装脚本的目的是直接从Dependencies目录中找到依赖包进行安装，否则可能会被应用商店卡住

理论上只有第一次安装需要使用脚本方式安装，但是难讲以后有依赖项变更时会不会出现问题

4.6.x更新4.7.x需要这么复杂是因为原本的编译选项无法编译所以修改了编译选项，这个已经找到可能的问题原因还没有确定

**Reply by WcZaHr** [reply]
好的，谢谢
